holidays/calendars/thai.py

```diff
@@ -21,76 +21,78 @@
+            4th Waning Day of Month 6
+            (On Month 7 for Athikamat Years).
+            Defaults to KHMER_CALENDAR (its sole user).
+        - Buddha's Cremation Day/Atthami Bucha:
+            8th Waning Day of Month 6
+            (On Month 7 for Athikamat Years).
+            KHMER_CALENDAR always falls on Month 6.
         - Asalha Puja/Asarnha Bucha:
-                15th Waxing Day (Full Moon) of Month 8
-                (On Month 8/8 for Athikamat Years).
-                KHMER_CALENDAR always fall on Month 8.
+            15th Waxing Day (Full Moon) of Month 8
+            (On Month 8/8 for Athikamat Years).
         - Buddhist Lent Day/Wan Khao Phansa:
-                1st Waning Day of Month 8
-                (On Month 8/8 for Athikamat Years).
-                KHMER_CALENDAR always fall on Month 8.
+            1st Waning Day of Month 8
+            (On Month 8/8 for Athikamat Years).
+        - Boun Haw Khao Padapdin/Boon Khao Padap Din:
+            14th Waning Day (New Moon) of Month 9.
+        - Boun Haw Khao Salark/Boon Khao Sak:
+            15th Waxing Day (Full Moon) of Month 10.
         - Pchum Ben/Prachum Bandar:
-                15th Waning Day (New Moon) of Month 10.
+            15th Waning Day (New Moon) of Month 10.
+        - Ok Boun Suang Huea/Vientiane Boat Racing Festival:
+            1st Waning Day (New Moon) of Month 11.
         - Loy Krathong/Boun That Louang/Bon Om Touk:
-                15th Waxing Day (Full Moon) of Month 12.
+            15th Waxing Day (Full Moon) of Month 12.
 
-    Other Buddhist date on Thai Lunar Calendar:
-        - Buddha's Cremation Day/Atthami Bucha
-                8th Waning Day of  Month 6
-                (On Month 7 for Athikamat Years).
-                KHMER_CALENDAR always fall on Month 6
+    Other Thai Lunar Calendar holidays:
+        - Thai Royal Ploughing Ceremony/Raeknakhwan:
+            Court astrologers choose the auspicious dates based on the Thai Lunar Calendar,
+            but these dates do not follow a predictable pattern.
+            See the specific section in `thailand.py` for more details.
         - End of Buddhist Lent Day/Ok Phansa:
-                15th Waxing Day (Full Moon) of Month 11
+            15th Waxing Day (Full Moon) of Month 11
+            (Currently calculated based on Asalha Puja/Asarnha Bucha method).
 
     The following code is based on Ninenik Narkdee's PHP implementation,
     and we're thankful for his work.
 
-    Please avoid touching the Athikawan and Athikamat declaration array
-    at all costs unless you can find sources for them somewhere for 2057++
-
-    Sources: (Ninenik.com 's wbm) http://tiny.cc/wa_ninenik_thluncal_php
-             https://www.myhora.com/ปฏิทิน/ปฏิทิน-พ.ศ.2560.aspx
+    References:
+        * <https://www.ninenik.com/แนวทางฟังก์ชั่น_php_อย่างง่ายกับการหาวันข้างขึ้นข้างแรม-1021.html>
+        * <https://www.myhora.com/ปฏิทิน/ปฏิทิน-พ.ศ.2560.aspx>
 
     Usage example:
```

Comment:
Code should be indented to be recognized as code:
```python
        >>> from holidays.calendars.thai import _ThaiLunisolar
        >>> thls = _ThaiLunisolar()
        >>> print(thls.visakha_bucha_date(2010))
        2010-05-28
```
(I can't add suggestion here)